From 8c829fdc37486abd0dac2729020bcdeb33f25e77 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Fran=C3=A7ois=20Laignel?= <francois@centricular.com>
Date: Fri, 23 Feb 2024 15:20:41 +0100
Subject: [PATCH] sdp: accept empty attribute value represented as a NULL
 pointer

Some empty media attribute values are set to an empty string, others as a NULL
pointer. It seems that code is able to deal with both, except for the UTF8
validation.

Part-of: <https://gitlab.freedesktop.org/gstreamer/gstreamer/-/merge_requests/6119>
---
 subprojects/gst-plugins-base/gst-libs/gst/sdp/gstsdpmessage.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/subprojects/gst-plugins-base/gst-libs/gst/sdp/gstsdpmessage.c b/subprojects/gst-plugins-base/gst-libs/gst/sdp/gstsdpmessage.c
index a08bbc96d1..96ed5fd5a2 100644
--- a/subprojects/gst-plugins-base/gst-libs/gst/sdp/gstsdpmessage.c
+++ b/subprojects/gst-plugins-base/gst-libs/gst/sdp/gstsdpmessage.c
@@ -4278,7 +4278,7 @@ sdp_add_attributes_to_caps (GArray * attributes, GstCaps * caps)
         continue;
 
       /* string must be valid UTF8 */
-      if (!g_utf8_validate (attr->value, -1, NULL))
+      if (attr->value != NULL && !g_utf8_validate (attr->value, -1, NULL))
         continue;
 
       if (!g_str_has_prefix (key, "x-"))
-- 
2.47.1

